src/deadline/job_attachments/api/manifest.py

```diff
@@ -54,72 +53,91 @@ def _manifest_snapshot(
         # Default, include all.
         glob_config = GlobConfig()
 
-    inputs = _glob_paths(root, include=glob_config.include_glob, exclude=glob_config.exclude_glob)
-
-    # Placeholder Asset Manager
-    asset_manager = S3AssetManager(
-        farm_id=" ", queue_id=" ", job_attachment_settings=JobAttachmentS3Settings(" ", " ")
+    current_files = _glob_paths(
+        root, include=glob_config.include_glob, exclude=glob_config.exclude_glob
     )
 
-    hash_callback_manager = _ProgressBarCallbackManager(length=100, label="Hashing Attachments")
-
-    upload_group = asset_manager.prepare_paths_for_upload(
-        input_paths=inputs, output_paths=[root], referenced_paths=[]
-    )
-    # We only provided 1 root path, so output should only have 1 group.
-    assert len(upload_group.asset_groups) == 1
-
-    if upload_group.asset_groups:
-        _, manifests = _hash_attachments(
-            asset_manager=asset_manager,
-            asset_groups=upload_group.asset_groups,
-            total_input_files=upload_group.total_input_files,
-            total_input_bytes=upload_group.total_input_bytes,
-            print_function_callback=logger.echo,
-            hashing_progress_callback=hash_callback_manager.callback,
+    # Compute the output manifest immediately and hash.
+    if not diff:
+        output_manifest = _create_manifest_for_single_root(
+            files=current_files, root=root, logger=logger
         )
-
-    if not manifests or len(manifests) == 0:
-        logger.echo("No manifest generated")
-        return None
-
-    # This is a hard failure, we are snapshotting 1 directory.
-    assert len(manifests) == 1
-    output_manifest = manifests[0].asset_manifest
-    if output_manifest is None:
-        raise ManifestCreationException()
+        if not output_manifest:
+            return None
 
     # If this is a diff manifest, load the supplied manifest file.
-    if diff:
+    else:
         # Parse local manifest
         with open(diff) as source_diff:
             source_manifest_str = source_diff.read()
             source_manifest = decode_manifest(source_manifest_str)
 
         # Get the differences
         changed_paths: List[str] = []
-        differences: List[Tuple[FileStatus, BaseManifestPath]] = compare_manifest(
-            source_manifest, output_manifest
-        )
-        for diff_item in differences:
-            if diff_item[0] == FileStatus.MODIFIED or diff_item[0] == FileStatus.NEW:
-                full_diff_path = f"{root}/{diff_item[1].path}"
-                changed_paths.append(full_diff_path)
-                logger.echo(f"Found difference at: {full_diff_path}, Status: {diff_item[0]}")
+
+        if fast_diff:
```

Comment:
nit: This logic is too long. It would be better to modularize into functions, or to follow a similar approach to make this section simpler.

Reply:
Agreed, let me move this out to a "fast_diff" function.